src/app/compare/Compare.tsx

```diff
@@ -185,6 +191,121 @@ class Compare extends React.Component<Props, State> {
 
     const stats = this.getAllStatsSelector(this.state, this.props);
 
+    // TODO: Should this go into its own module?
+    const updateSocketComparePlug = ({
+      item,
+      categoryHash,
+      socket,
+      plug: clickedPlug,
+    }: {
+      item: DimItem;
+      categoryHash: number;
+      socket: DimSocket;
+      plug: DimPlug;
+    }) => {
+      // Exit early if this is a D1 item or an item without sockets
+      if (!item.isDestiny2() || !item.sockets || item.sockets === undefined) {
+        return null;
+      }
+
+      let { socketIndex } = socket;
+      const categoryIndex = item.sockets.categories.findIndex(
+        (category) => category.category.hash === categoryHash
+      );
+      // Create deep copies of arrays to prevent directly mutating state
+      // (cloneDeep is resulting in object mutation)
+      const itemStatsClone: DimStat[] = JSON.parse(JSON.stringify(item.stats));
+
+      // socketIndex is correct for allSockets, but not categories[].sockets
+      // TODO: Confirm socket is always off by 1, otherwise this should use hashes to find plugs
```

Comment:
The fact that the indexes are off by 1 makes me a little nervous about this approach especially if its a bungie api bug. For that reason if we can move to a hash based approach I think it would be the way to go for safety.

Reply:
Turns out this was a hangover from storing references to the actual plug in state. Don't need to use `categories[].sockets` any more.